dashboard/src/shared/ResourceRef.ts

```diff
@@ -0,0 +1,43 @@
+import { Kube } from "./Kube";
+import { IResource } from "./types";
+
+class ResourceRef {
+  // Creates a new ResourceRef instance from an existing IResource. Provide
+  // defaultNamespace to set if the IResource doesn't specify a namespace.
+  // TODO: add support for cluster-scoped resources, or add a ClusterResourceRef
+  // class.
+  public static newFromResource(r: IResource, defaultNamespace: string = "default") {
```

Comment:
why isn't this a proper class constructor? This sounds GO idiomatic.

Reply:
What if we want to create one without an IResource, in tests for example?

Reply:
the pattern we have been following is to create a dummy object with the required elements that we reuse ass "default" props

Reply:
I think the issue is that that approach doesn't work for classes because it requires the `getResourceURL` to be present in that dummy object, but let me try again.

Reply:
I switched this to a proper constructor - thanks!